src/core/ext/transport/chaotic_good/server/chaotic_good_server.cc

```diff
@@ -141,11 +221,69 @@ absl::StatusOr<int> ChaoticGoodServerListener::Bind(
   return port_num;
 }
 
+absl::Status ChaoticGoodServerListener::BindExternal(std::string addr,
+                                                     const ChannelArgs& args) {
+  using grpc_event_engine::experimental::EventEngine;
+  class FdHandler final : public TcpServerFdHandler {
+   public:
+    FdHandler(RefCountedPtr<ChaoticGoodServerListener> listener,
+              grpc_event_engine::experimental::ListenerSupportsFdExtension*
+                  listener_supports_fd)
+        : listener_(std::move(listener)),
+          listener_supports_fd_(listener_supports_fd) {}
+
+    virtual void Handle(int listener_fd, int fd,
+                        grpc_byte_buffer* pending_read) {
+      grpc_event_engine::experimental::SliceBuffer pending_data;
+      if (pending_read != nullptr) {
+        pending_data =
+            grpc_event_engine::experimental::SliceBuffer::TakeCSliceBuffer(
+                pending_read->data.raw.slice_buffer);
+      }
+      std::string pending_data_str;
+      for (size_t i = 0; i < pending_data.Count(); ++i) {
+        absl::StrAppend(&pending_data_str, pending_data[i].as_string_view());
```

Comment:
This seems like it could be a bit expensive - should we put this under chaotic good tracing? And is there any PII issue?

Reply:
oh yeah we should drop that